app/services/container_topology_service.rb

```diff
@@ -61,14 +61,15 @@ def entity_status(entity)
     if entity.kind_of?(Host) || entity.kind_of?(Vm)
       status = entity.power_state.capitalize
     elsif entity.kind_of?(ContainerNode)
-      status = 'Unknown'
-      entity.container_conditions.each do |condition|
-        if condition.try(:name) == 'Ready' && condition.try(:status) == 'True'
-          status = condition.name
-        else
-          status = 'NotReady'
-        end
-      end
+      condition_status = entity.container_conditions.find_by_name('Ready').try(:status)
+      status = case condition_status
```

Comment:
@zeari technically that could become a one-liner. If you prefer to keep them separated then maybe we can use a name more descriptive variable name e.g. `node_ready_status` or `node_is_ready` (True/False).